### PR TITLE
feat: add cancellation to Deploy Retrieve Library commands

### PR DESCRIPTION
**Pull request by sfsholden**
### What does this PR do?
Adds cancellation logic for Deploy Retrieve Library commands.

### What issues does this PR fix or reference?
@W-8979870@

### Functionality Before
Before

### Functionality After
SDR_After

### CODE REVIEW

File: packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts [hunk 1]
```diff
@@ -120,6 +121,38 @@ describe('LibraryCommandletExecutor', () => {
...
+
+    await cancelledExecutor.execute({ data: { success: true }, type: 'CONTINUE' });
+
+    expect(withProgressStub.called).to.be.true;
+    expect(withProgressStub.getCall(0).args[0]).to.eql({
+      title: nls.localize('progress_notification_text', 'Test Command'),
+      location: vscodeStub.ProgressLocation.Notification,
+      cancellable: true
+    });
+    // expect(cancelStub.calledOnce).to.be.true;
```
Comment (sfsholden) [inline packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts:151]: Still incomplete, but I realized that some of the tests I was trying to build within baseDeployRetrieve weren't achievable and made more sense to have here anyways. I have the mocked token and I am deliberately trying to invoke cancel from it. However, the `token.onCancellationRequested(()` still doesn't fire here. I feel this is close - any ideas on what is missing?

File: packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts [hunk 2]
```diff
@@ -40,15 +43,26 @@ import { createComponentCount } from './util';
...
 type RetrieveResult = MetadataApiRetrieveResult | SourceRetrieveResult;
 type DeployRetrieveResult = DeployResult | RetrieveResult;
+type DeployRetrieveOperation = MetadataApiDeploy | MetadataApiRetrieve;
 
 export abstract class DeployRetrieveExecutor<
   T
-> extends LibraryCommandletExecutor<T> {
+  > extends LibraryCommandletExecutor<T> {
+
+  protected cancellable: boolean = true;
```
Comment (sfsholden) [inline packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts:52]: This property is what shows the "Cancel" button on the commands.

File: packages/salesforcedx-vscode-core/test/vscode-integration/commands/baseDeployRetrieve.test.ts [hunk 3]
```diff
@@ -490,6 +508,15 @@ describe('Base Deploy Retrieve Commands', () => {
       expect(executor.retrieveStub.callCount).to.equal(1);
     });
 
+    it('should call setup cancellation logic', async () => {
```
Comment (sfsholden) [inline packages/salesforcedx-vscode-core/test/vscode-integration/commands/baseDeployRetrieve.test.ts:511]: I wanted to verify that a cancel command would call the .cancel method on the operation. However I wasn't able to figure out how to do this within the baseDeployRetrieve tests.

File: packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts (hunk 1 above)
Comment (sfsholden) [reply packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts:151]: I will give that a shot - thank you

Comment (sfsholden) [reply packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts:151]: @xyc This did the trick. Thank you so much!

**Review by AnanyaJha — APPROVED** [review]

**Pull request merged**